hfile.c

```diff
@@ -573,6 +573,9 @@ static hFILE *hopen_fd(const char *filename, const char *mode)
 
 hFILE *hdopen(int fd, const char *mode)
 {
+#if defined HAVE_SETMODE && defined O_BINARY
+    if (setmode(fd, O_BINARY) < 0) return NULL;
+#endif
```

Comment:
Relevant part from #283 (quoting @jmarshall):
>I would prefer not to call `setmode()` in `hdopen()` because:
>
 >  1. `hdopen()` can be called with file descriptors that are not files. In particular, on Windows sockets will not respond to `setmode()` particularly well;
>   2. file descriptors given to `hdopen()` may already have had I/O done on them, so it may by then be too late to call `setmode()`.


So we should fix test_bgzf.c, and note in the documentation for `hdopen()` that the file descriptor passed in must already be in binary mode.